### PR TITLE
Rtti reflection, major bump

### PR DESCRIPTION
**Pull request by capaj**
implements RTTI reflection FTW

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,18 +124,17 @@ class SuperSchema {
...
-There are, however cases where we have to define them explicitly.
+- Unions - for example we want to specify whether field is nullable or not
+- Function returns type of `Promise<SomeType>`
+- List (Array) type is used
 
-- We want to specify whether field is nullable or not
-- We want to be explicit about if some `number` type is `Float` or `Int` (`GraphQLFloat` or `GraphQLInt`) etc
-- Function we use returns type of `Promise<SomeType>` while field itself is typed as `SomeType`
-- List (Array) type is used. (For now, typescript `Reflect` api is not able to guess type of single array item)
+All other code-first libraries on decorators like typegraphql or typegql require you to write types for these twice. Decapi infers is from typescript without any extra effort on your end.
```
Comment (Misticek) [inline README.md]:
```suggestion
All other code-first libraries on decorators like typegraphql or typegql require you to write types for these twice. Decapi infers types from typescript without any extra effort on your end.
```

File: docs/explore/duplex-type-and-duplex-field.mdx [hunk 2]
```diff
@@ -19,7 +19,7 @@ class User {
   @Field()
   password_hash: string
 
-  @DuplexArrayField({ itemType: String})
+  @DuplexField({ itemType: String })
```
Comment (Misticek) [inline docs/explore/duplex-type-and-duplex-field.mdx]: If decapi 2 is able to infer type from ts, why is `String` type explicitly used here?

Comment (capaj) [reply docs/explore/duplex-type-and-duplex-field.mdx]: I haven't updated the documentation yet 😞

File: src/specs/mutation.spec.ts [hunk 3]
```diff
@@ -67,7 +67,7 @@ describe('Mutation', () => {
         }
       `
     })
-
+    console.error(result.errors)
```
Comment (Misticek) [inline src/specs/mutation.spec.ts]: I think you don't wanna console it out as you are asserting it after that.

File: src/specs/inputObjectTypeAsInstance.spec.ts [hunk 4]
```diff
@@ -72,7 +72,7 @@ describe('input object type arguments are passed as instances', () => {
         }
       `
     })
-
+    console.error(result1.errors)
```
Comment (Misticek) [inline src/specs/inputObjectTypeAsInstance.spec.ts]: 🍪

File: package.json [hunk 5]
```diff
@@ -41,37 +41,43 @@
...
   "devDependencies": {
     "@types/graphql": "^14.5.0",
-    "@types/jest": "^27.4.0",
-    "@types/node": "^17.0.16",
+    "@types/jest": "^27.4.1",
+    "@types/node": "^17.0.23",
     "@types/object-path": "^0.11.1",
     "@types/reflect-metadata": "^0.1.0",
-    "docz": "^2.3.1",
+    "docz": "^1",
```
Comment (Misticek) [inline package.json]: Wrong version

Comment (capaj) [reply package.json]: oh I had to revert this. I bumped it to major 2 by mistake-docz 2 is not compatible with docz 1 config and I've never migrated the config.
Version 1 should work fine.

**Comment by capaj** [comment]
There is one last thing before the release I want to do-make enums registration automatic. Curently user has to call `registerEnum`. ~With rtti, we can do this automatically.~
Then we will push this into 2.0.0

EDIT: not really-we still need  to register the enum, but we can use it without using explicit `type`

**Pull request merged**